File: CodeWars/JavaScript/7Kyu/FindTheLongestGap.js
// URL: https://www.codewars.com/kata/find-the-longest-gap/train/javascript

function gap(num) {
  maxLengthFound = 0;
  string = "";

  binary = num.toString(2);

  for (i=0;i<binary.length;i++){
    if (binary[i] == "1"){
      if (maxLengthFound<string.length){
        maxLengthFound = string.length;
      }
        string = "";
    }
    else{
      string += binary[i];
    }
  }  

  return maxLengthFound;

}

console.log(gap(15));

File: CodeWars/Python/7Kyu/FindTheLongestGap.py
# URL: https://www.codewars.com/kata/find-the-longest-gap/train/python

def gap(num):
  maxLengthFound = 0
  stringLoop = ""

  binary = bin(num)[2:];

  for number in binary:
    if (number == "1"):
      if (maxLengthFound<len(stringLoop)):
        maxLengthFound = len(stringLoop)
      stringLoop = ""
    else:
      stringLoop += number 

  return maxLengthFound

print(gap(15))